lib/PuppeteerSharp/Frame.cs

```diff
@@ -69,15 +70,25 @@ internal Frame(FrameManager frameManager, CDPSession client, Frame parentFrame,
 
             if (parentFrame != null)
             {
-                ParentFrame.ChildFrames.Add(this);
+                lock (ParentFrame._childFrames) {
+                    ParentFrame._childFrames.Add(this);
+                }
             }
         }
 
         #region Properties
+        internal List<Frame> _childFrames { get; } = new List<Frame>();
+
```

Comment:
We could leave only one public property and add an internal `AddChildFrame` method. There we can do the lock.

Reply:
We need the lock for both reading and writing though.

Reply:
Sure. But I don't like that internal prop

Reply:
We need a backing collection declared somewhere and `ChildFrames` is a public api, so we need both.

Reply:
Even changing the declaration of `ChildFrames` from `List<Frame>` to `IList<Frame>` would be a breaking change.

Reply:
We could factor this into out into a `ConcurrentList` but I remember you saying you would want to avoid that.

Reply:
We could have that but as internal variable instead a property. And let the child frames interact with that `_childFrames` variable through an internal method.